source/shadow.c

```diff
@@ -297,6 +398,41 @@ static ShadowStatus_t validateName( const char * pString,
 
 /*-----------------------------------------------------------*/
 
+static ShadowStatus_t extractThingName( const char * pTopic,
+                                        uint16_t topicLength,
+                                        uint16_t * pConsumedTopicLength,
+                                        uint16_t * pThingNameLength )
+{
+    ShadowStatus_t shadowStatus;
```

Comment:
maintainability: initialize `shadowStatus` here, e.g.:

```c
ShadowStatus_t shadowStatus = validateName(.  ...
```

If that isn't permissible due to coding standard or static analysis, then just do:

```c
ShadowStatus_t shadowStatus = SHADOW_BAD_PARAMETER;
```

There will be no run-time impact, the compiler will eliminate any redundant assignment – however, the code will be more resilient to mishandling of the `shadowStatus` variable in future revisions.